---
title: "Diversity"
output: html_document
date: "2024-06-04"
---
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
#Data Preprocessing#

## Libraries ##
```{r}
library(BiocManager)
library(phyloseq)
library(vegan)
library(tidyverse)
library(metagenomeSeq)
library(ggpubr)
library(Biostrings)
library(microbiome)
library(ggrepel)
library(decontam)
library(picante) #For Faiths Phylogentic Diversity
```
## Set global options ##
```{r}
# Set global options #
# no scientific notation
options(scipen=10000) 

# color blind pallets used throughout 
cbbPalette <- c("#000000", "#E69F00", "#56B4E9", "#009E73", "#F0E442", "#0072B2", "#D55E00", "#CC79A7")
ibm.cbb <- c("#648FFF", "#785EF0", "#DC267F", "#FE6100", "#FFB000")
tol.cbb <- c("#332288", "#117733", "#44AA99", "#88CCEE", "#DDCC77", "#CC6677", "#AA4499", "#882255")
```

##NOTE: Rmd files set the working directory to the place where the rmd file is, not the actual working directory
##Read in and Separation##
## Read in the RDS files #
```{r}
#Fungi RDS
FSpermosphere_RDS_CSS <- readRDS("2024-02-08_SpermosphereSteaming_Fungi/RDS/Fungi_spermosphere_CSS_2024-03-13.rds") ##Using Normalized Reads
FSpermosphere_RDS_NoNorm <- readRDS("2024-02-08_SpermosphereSteaming_Fungi/RDS/Fungi_spermosphere_nonorm_2024-03-13.rds") ##Using Non-normalized Reads
FSpermosphere_RDS_Rareified <- readRDS("2024-02-08_SpermosphereSteaming_Fungi/RDS/Fungi_spermosphere_Rarefied_2024-06-04.rds") ##Using Rarefied Reads w/o Replacement
#Bacteria RDS 
BSpermosphere_RDS_CSS <- readRDS("2024-01-24_SpermosphereSteaming_Bacteria/RDS/Bacteria_spermosphere_nonnorm_CSS_2024-01-24.rds") ##Using Normalized Reads
BSpermosphere_RDS_NoNorm <- readRDS("2024-01-24_SpermosphereSteaming_Bacteria/RDS/Bacteria_spermosphere_nonnorm_2024-01-24.rds") ##Using Non-normalized Reads
BSpermosphere_RDS_Rareified <- readRDS("2024-01-24_SpermosphereSteaming_Bacteria/RDS/Bacteria_spermosphere_Rarefied_2024-06-06.rds") ##Using Rarefied Reads w/o Replacement
```

##Seperation of Nonsteamed vs Steamed soils##
```{r}
##There is a space after Steamed in the Metadata File
fun.Steamed <- subset_samples(FSpermosphere_RDS_NoNorm, Trt == "Steamed ")
fun.Nonsteamed <- subset_samples(FSpermosphere_RDS_NoNorm, Trt == "Non.Steamed")
bac.Steamed <- subset_samples(BSpermosphere_RDS_NoNorm, Trt == "Steamed ")
bac.Nonsteamed <- subset_samples(BSpermosphere_RDS_NoNorm, Trt == "Non.Steamed")

```
## separating the otu table, metadata, and the tax data for ease #
```{r}
#FUNGI Steamed
Sfun.meta.data <- data.frame(fun.Steamed@sam_data)
Sfun.tax.data <- data.frame(fun.Steamed@tax_table)
Sfun.otu <- fun.Steamed@otu_table %>%
  as("matrix")
#FUNGI NonSteamed
Nfun.meta.data <- data.frame(fun.Nonsteamed@sam_data)
Nfun.tax.data <- data.frame(fun.Nonsteamed@tax_table)
Nfun.otu <- fun.Nonsteamed@otu_table %>%
  as("matrix")
#BACTERIA Steamed
Sbac.meta.data <- data.frame(bac.Steamed@sam_data)
Sbac.tax.data <- data.frame(bac.Steamed@tax_table)
Sbac.otu <- bac.Steamed@otu_table %>%
  as("matrix")
#BACTERIA NonSteamed
Nbac.meta.data <- data.frame(bac.Nonsteamed@sam_data)
Nbac.tax.data <- data.frame(bac.Nonsteamed@tax_table)
Nbac.otu <- bac.Nonsteamed@otu_table %>%
  as("matrix")

cssbac.meta.data <- data.frame(BSpermosphere_RDS_CSS@sam_data)
cssfun.meta.data <- data.frame(FSpermosphere_RDS_CSS@sam_data)
```
#Data Analysis#
##Abundance Occupancy
###Abundance occupancy Fungi Steamed Soil##
```{r}
Sfun.otu_PA <- 1*((Sfun.otu>0)==1)                                               # presence-absence data
Sfun.otu_occ <- rowSums(Sfun.otu_PA)/ncol(Sfun.otu_PA)                                # occupancy calculation
Sfun.otu_rel <- apply(decostand(Sfun.otu, method="total", MARGIN=2),1, mean)     # mean relative abundance
Sfun.occ_abun <- rownames_to_column(as.data.frame(cbind(Sfun.otu_occ, Sfun.otu_rel)),'OTU')
colnames(Sfun.occ_abun) = c("OTU", "fun.otu_occ", "fun.otu_rel")
Sfun.occ_abund_tax <- left_join(Sfun.occ_abun, Sfun.tax.data, by = "OTU")

Sfun.occ_abund_tax$Other <- ifelse(Sfun.occ_abund_tax$fun.otu_occ < 0.80 & Sfun.occ_abund_tax$fun.otu_rel < 0.05, "Other", Sfun.occ_abund_tax$Order)
```
####incase we need to use this as its own figure.
```{r}
fig1a <- ggplot(Sfun.occ_abund_tax, aes(y = fun.otu_occ, x = log10(fun.otu_rel), color = Other)) +
  geom_point() + 
  theme_classic2() +
  #scale_fill_manual(values= c(cbbPalette, ibm.cbb, tol.cbb, "red")) +
  scale_y_continuous(labels = scales::percent) +
  #scale_x_continuous(labels = scales::percent) + 
  xlab("Log10(Relative Abundance)") +
  ylab("% Occupancy") + 
  scale_color_manual(values = c(cbbPalette, ibm.cbb, tol.cbb)) +
  geom_text_repel(data = Sfun.occ_abund_tax[Sfun.occ_abund_tax$otu_occ > 0.30 | Sfun.occ_abund_tax$otu_rel > 0.05,], 
                  aes(y = Sfun.otu_occ, x = Sfun.otu_rel, color = Other, label = Label), size = 2) +
  theme(legend.text = element_text(face = "italic", size = 8),
legend.title = element_blank())
fig1a
```

###Abundance occupancy Fungi NonSteamed Soil#
```{r}
Nfun.otu_PA <- 1*((Nfun.otu>0)==1)                                               # presence-absence data
Nfun.otu_occ <- rowSums(Nfun.otu_PA)/ncol(Nfun.otu_PA)                                # occupancy calculation
Nfun.otu_rel <- apply(decostand(Nfun.otu, method="total", MARGIN=2),1, mean)     # mean relative abundance
Nfun.occ_abun <- rownames_to_column(as.data.frame(cbind(Nfun.otu_occ, Nfun.otu_rel)),'OTU')
colnames(Nfun.occ_abun) = c("OTU", "fun.otu_occ", "fun.otu_rel")
Nfun.occ_abund_tax <- left_join(Nfun.occ_abun, Nfun.tax.data, by = "OTU")

Nfun.occ_abund_tax$Other <- ifelse(Nfun.occ_abund_tax$fun.otu_occ < 0.80 & Nfun.occ_abund_tax$fun.otu_rel < 0.05, "Other", Nfun.occ_abund_tax$Order)

Sfun.occ_abund_tax$Steamed2 <- "Steamed"
Nfun.occ_abund_tax$Steamed2 <- "Non-Steamed"

FCombined <- rbind.data.frame(Sfun.occ_abund_tax, Nfun.occ_abund_tax)

fig1a <- ggplot(FCombined, aes(y = fun.otu_occ, x = log10(fun.otu_rel), color = Other)) +
  geom_point() + 
  theme_classic2() + 
  #transition_states(Steamed2, transition_length = 5, state_length = 5) +
  #scale_fill_manual(values= c(cbbPalette, ibm.cbb, tol.cbb)) +
  scale_y_continuous(labels = scales::percent) +
  #scale_x_continuous(labels = scales::percent) + 
  geom_hline(yintercept = .8, linetype = "dashed") +
  xlab("Log10(Relative Abundance)") +
  ylab("% Occupancy") + 
  scale_color_manual(values = c(ibm.cbb, cbbPalette, tol.cbb)) +
  #geom_text_repel(data = FCombined[FCombined$fun.otu_occ > 0.80 | FCombined$fun.otu_rel > 0.05,], 
                  #aes(y = fun.otu_occ, x = log10(fun.otu_rel), color = Other, label = Label), size = 2) +
  facet_wrap(~Steamed2) +
  theme(legend.text = element_text(face = "italic", size = 8), legend.title = element_blank())
fig1a
```

```{r}

fig1b <- ggplot(FCombined, aes(y = fun.otu_occ, x = log10(fun.otu_rel), color = Other)) +
  geom_point(size = 4) + 
  theme_classic2() + 
  #transition_states(Steamed2, transition_length = 5, state_length = 5) +
  #scale_fill_manual(values= c(cbbPalette, ibm.cbb, tol.cbb)) +
  scale_y_continuous(labels = scales::percent) +
  #scale_x_continuous(labels = scales::percent) + 
  geom_hline(yintercept = .8, linetype = "dashed") +
  xlab("Log10(Relative Abundance)") +
  ylab("% Occupancy") +
  coord_cartesian(ylim = c(0.79, 1.01), expand = FALSE) +
  scale_color_manual(values = c(ibm.cbb, cbbPalette, tol.cbb)) +
  geom_text_repel(data = FCombined[FCombined$fun.otu_occ > 0.80 | FCombined$fun.otu_rel > 0.05,], 
                  aes(y = fun.otu_occ, x = log10(fun.otu_rel), color = Other, label = Label), size = 3.5) +
  facet_wrap(~Steamed2) +
  theme(legend.text = element_text(face = "italic", size = 8), legend.title = element_blank())
fig1b
```

###Abundance occupancy Bacteria Steamed Soil#
```{r}
Sbac.otu_PA <- 1*((Sbac.otu>0)==1)                                               # presence-absence data
Sbac.otu_occ <- rowSums(Sbac.otu_PA)/ncol(Sbac.otu_PA)                                # occupancy calculation
Sbac.otu_rel <- apply(decostand(Sbac.otu, method="total", MARGIN=2),1, mean)     # mean relative abundance
Sbac.occ_abun <- rownames_to_column(as.data.frame(cbind(Sbac.otu_occ, Sbac.otu_rel)),'OTU')
colnames(Sbac.occ_abun) = c("OTU", "bac.otu_occ", "bac.otu_rel")
Sbac.occ_abund_tax <- left_join(Sbac.occ_abun, Sbac.tax.data, by = "OTU")

Sbac.occ_abund_tax$Other <- ifelse(Sbac.occ_abund_tax$bac.otu_occ < 0.96 & Sbac.occ_abund_tax$bac.otu_rel < 0.05, "Other", Sbac.occ_abund_tax$Phylum)


```

###Abundance occupancy Bacteria NonSteamed Soil#

```{r}
Nbac.otu_PA <- 1*((Nbac.otu>0)==1)                                               # presence-absence data
Nbac.otu_occ <- rowSums(Nbac.otu_PA)/ncol(Nbac.otu_PA)                                # occupancy calculation
Nbac.otu_rel <- apply(decostand(Nbac.otu, method="total", MARGIN=2),1, mean)     # mean relative abundance
Nbac.occ_abun <- rownames_to_column(as.data.frame(cbind(Nbac.otu_occ, Nbac.otu_rel)),'OTU')
colnames(Nbac.occ_abun) = c("OTU", "bac.otu_occ", "bac.otu_rel")
Nbac.occ_abund_tax <- left_join(Nbac.occ_abun, Nbac.tax.data, by = "OTU")

Nbac.occ_abund_tax$Other <- ifelse(Nbac.occ_abund_tax$bac.otu_occ < 0.96 & Nbac.occ_abund_tax$bac.otu_rel < 0.05, "Other", Nbac.occ_abund_tax$Phylum)

Sbac.occ_abund_tax$Steamed2 <- "Steamed"
Nbac.occ_abund_tax$Steamed2 <- "Non-Steamed"

BCombined <- rbind.data.frame(Sbac.occ_abund_tax, Nbac.occ_abund_tax)

fig2a <- ggplot(BCombined, aes(y = bac.otu_occ, x = log10(bac.otu_rel), color = Other)) +
  geom_point() + 
  theme_classic2() + 
  #transition_states(Steamed2, transition_length = 5, state_length = 5) +
  #scale_fill_manual(values= c(cbbPalette, ibm.cbb, tol.cbb)) +
  scale_y_continuous(labels = scales::percent) +
  #scale_x_continuous(labels = scales::percent) + 
  geom_hline(yintercept = .96, linetype = "dashed") +
  xlab("Log10(Relative Abundance)") +
  ylab("% Occupancy") + 
  scale_color_manual(values = c(ibm.cbb, cbbPalette, tol.cbb, "red", "black")) +
  #geom_text_repel(data = FCombined[FCombined$fun.otu_occ > 0.80 | FCombined$fun.otu_rel > 0.05,], 
                  #aes(y = fun.otu_occ, x = log10(fun.otu_rel), color = Other, label = Label), size = 2) +
  facet_wrap(~Steamed2) +
  theme(legend.text = element_text(face = "italic", size = 8), legend.title = element_blank())
fig2a
```

```{r}
fig2b <- ggplot(BCombined, aes(y = bac.otu_occ, x = log10(bac.otu_rel), color = Other)) +
  geom_point() + 
  theme_classic2() + 
  #transition_states(Steamed2, transition_length = 5, state_length = 5) +
  #scale_fill_manual(values= c(cbbPalette, ibm.cbb, tol.cbb)) +
  scale_y_continuous(labels = scales::percent) +
  #scale_x_continuous(labels = scales::percent) + 
  geom_hline(yintercept = .96, linetype = "dashed") +
  xlab("Log10(Relative Abundance)") +
  ylab("% Occupancy") +
  coord_cartesian(ylim = c(0.95, 1.01), expand = FALSE) +
  scale_color_manual(values = c(ibm.cbb, cbbPalette, tol.cbb, "red", "black")) +
  geom_text_repel(data = BCombined[BCombined$bac.otu_occ > 0.96 | BCombined$bac.otu_rel > 0.05,], 
                  aes(y = bac.otu_occ, x = log10(bac.otu_rel), color = Other, label = Genus), size = 3.5) +
  facet_wrap(~Steamed2) +
  theme(legend.text = element_text(face = "italic", size = 8), legend.title = element_blank())
fig2b

```

###Abundance occupancy Template#
```{r}
otu_PA <- 1*((otu>0)==1)                                               # presence-absence data
otu_occ <- rowSums(otu_PA)/ncol(otu_PA)                                # occupancy calculation
otu_rel <- apply(decostand(otu, method="total", MARGIN=2),1, mean)     # mean relative abundance
occ_abun <- rownames_to_column(as.data.frame(cbind(otu_occ, otu_rel)),'OTU')

occ_abund_tax <- left_join(occ_abun, tax.data, by = "OTU")

occ_abund_tax$Other <- ifelse(occ_abund_tax$otu_occ < 0.31 & occ_abund_tax$otu_rel < 0.05, "Other", occ_abund_tax$Genus)

fig1a <- ggplot(occ_abund_tax, aes(y = otu_occ, x = otu_rel, color = Other)) +
  geom_point() + 
  theme_classic2() + 
  scale_fill_manual(values= c(cbbPalette, ibm.cbb, tol.cbb)) +
  scale_y_continuous(labels = scales::percent) +
  scale_x_continuous(labels = scales::percent) + 
  xlab("% Relative Abundance") +
  ylab("% Occupancy") + 
  scale_color_manual(values = c(cbbPalette, ibm.cbb)) +
  geom_text_repel(data = occ_abund_tax[occ_abund_tax$otu_occ > 0.30 | occ_abund_tax$otu_rel > 0.05,], 
                  aes(y = otu_occ, x = otu_rel, color = Other, label = Label), size = 2) +
  theme(legend.text = element_text(face = "italic", size = 8),
legend.title = element_blank())
```

### Presence absence of otus#
```{r}
PA_table <- data.frame(t(otu_PA)) %>%
  rownames_to_column() %>%
  pivot_longer(cols = OOTU_103:OOTU_937, values_to = "PA", names_to = "OTU") %>%
  rename("rowname" = "Sample") %>%
  left_join(meta.data, by = "Sample") %>%
  left_join(tax.data, by = "OTU") %>%
  subset(Genus %in% c("Pythium",
                      "Globisporangium", 
                      "Phytophthora", 
                      "Phytopythium"))


fig1c <- ggplot(PA_table, aes(y = reorder(Label, PA), x = reorder(Sample.y.y, Latitude), fill = as.factor(PA))) +
  geom_tile(color = "black") + 
  theme_classic() + 
  scale_fill_manual(values = c("white",cbbPalette[[4]]), name = "", labels = c("Absent", "Present")) +
  theme(axis.text.x = element_text(angle=45, hjust=1, size = 5),
        axis.text.y = element_text(face = "italic", size = 5)) +
  xlab("") + 
  ylab("") 

```
#Alpha Diversity#
##Prokaryote Alpha Diversity##
###NonSteamed Prokaryote Alpha Diversity###
```{r}

bac.Nonsteamed@sam_data$shannon <- estimate_richness(bac.Nonsteamed, measures=c("Shannon"))$Shannon
bac.Nonsteamed@sam_data$invsimpson <- estimate_richness(bac.Nonsteamed, measures=c("InvSimpson"))$InvSimpson
bac.Nonsteamed@sam_data$simpson <- estimate_richness(bac.Nonsteamed, measures=c("Simpson"))$Simpson
bac.Nonsteamed@sam_data$richness <- estimate_richness(bac.Nonsteamed, measures=c("Observed"))$Observed
bac.Nonsteamed@sam_data$even <- bac.Nonsteamed@sam_data$shannon/log(bac.Nonsteamed@sam_data$richness)

Nsample.data.bac <- data.frame(bac.Nonsteamed@sam_data)

Nsample.data.bac$Time <- factor(Nsample.data.bac$Time, levels = c("17hrs", "Planting", "V2"))
```

```{r}
# Faith's phylogenetic diversity
bnonsteamed_otu <- bac.Nonsteamed@otu_table %>%
  as.data.frame()
nbphylo.diversity <- pd(t(bnonsteamed_otu), bac.Nonsteamed@phy_tree, include.root = F)
nbphylo.diversity$Samples <- rownames(nbphylo.diversity) 

nbphylo.diversity.join <- left_join(nbphylo.diversity, as.data.frame(bac.Nonsteamed@sam_data), by = c("Samples" = "Tube"))
nbphylo.diversity.join$Time.Point <- factor(nbphylo.diversity.join$Time, levels = c("Planting", "17hrs", "V2"))
####### Richness over time; supplemental figure part 1 ######
compare_means(richness ~ Time, Nsample.data.bac, group.by = "Type")
kruskal.test(richness ~ Time, Nsample.data.bac, group.by = "Type")
####### Faith's phylogenetic diversity over time; supplemental figure #######
compare_means(PD ~ Crop, phylo.diversity.join, group.by = "Time.Point")
compare_means(PD ~ Time.Point, phylo.diversity.join, method = "kruskal")
```
####### Bacterial evenness; Figure 2a ########
```{r}
compare_means(even ~ Crop, sample.data.bac, group.by = "Time.Point")
```
###### Supplemental fungal alpha diversity figure ####
```{r}
Supplemental.figure4 <- ggpubr::ggarrange(fungi.richness, 
                                          fungi.even,
                                          labels = "auto",
                                          nrow = 1, ncol = 2, common.legend = T)
```
###Steamed Prokaryote Alpha Diversity###
```{r}

bac.Steamed@sam_data$shannon <- estimate_richness(bac.Steamed, measures=c("Shannon"))$Shannon
bac.Steamed@sam_data$invsimpson <- estimate_richness(bac.Steamed, measures=c("InvSimpson"))$InvSimpson
bac.Steamed@sam_data$simpson <- estimate_richness(bac.Steamed, measures=c("Simpson"))$Simpson
bac.Steamed@sam_data$richness <- estimate_richness(bac.Steamed, measures=c("Observed"))$Observed
bac.Steamed@sam_data$even <- bac.Steamed@sam_data$shannon/log(bac.Steamed@sam_data$richness)

Ssample.data.bac <- data.frame(bac.Steamed@sam_data)

Ssample.data.bac$Time <- factor(Ssample.data.bac$Time, levels = c("17hrs", "Planting", "V2"))
```

```{r}
# Faith's phylogenetic diversity
bSteamed_otu <- bac.Steamed@otu_table %>%
  as.data.frame()
Sbphylo.diversity <- pd(t(bSteamed_otu), bac.Steamed@phy_tree, include.root = F)
Sbphylo.diversity$Samples <- rownames(Sbphylo.diversity) 

Sbphylo.diversity.join <- left_join(Sbphylo.diversity, as.data.frame(bac.Steamed@sam_data), by = c("Samples" = "Tube"))
Sbphylo.diversity.join$Time.Point <- factor(Sbphylo.diversity.join$Time, levels = c("17hrs", "Planting", "V2"))
```

###Conjoined Bacterial Alpha Diversity###
```{r}
Sbphylo.diversity.join$Steamed2 <- "Steamed"
nbphylo.diversity.join$Steamed2 <- "NonSteamed"

BAlphaCombined <- rbind.data.frame(Sbphylo.diversity.join, nbphylo.diversity.join)
```

####RICHNESS
```{r}
compare_means(richness ~ Trt, Sbphylo.diversity.join, group.by = "Time.Point")
compare_means(richness ~ Time, Ssample.data.bac)

bac.richness <- BAlphaCombined %>%
  ggplot(aes(x = Type, y = richness, color = Steamed2)) +
  facet_wrap(~factor(Time, levels = c("Planting", "17hrs", "V2")), scale = "free") +
  geom_boxplot(position = position_dodge2(0.85, preserve = "single")) + 
  geom_point(position=position_jitterdodge(0.05)) +
  scale_y_continuous(limits = c(0, 4000), breaks = seq(0, 4000, 500)) +
  stat_compare_means(method = "t.test") +
  scale_fill_manual(values = cbbPalette ) +
  color_palette(cbbPalette) +
  #stat_summary(fun = mean,geom="line") +
  #stat_summary(fun.data = mean_se, geom = "errorbar", width = 0.5) +
  ylab("Richness") +
  xlab("Type of Environment")+
  #geom_jitter(width = 0.1, alpha = 0.8)+
  #scale_color_manual(values=cbbPalette, name = "", labels = c("Planting", "17hrs", "V2")) +  
  #stat_compare_means(method = "kruskal", hide.ns = TRUE) +
  theme_bw()+
  theme(legend.text = element_text(face = "italic", size = 12), legend.title = element_blank())
bac.richness

RichWil <- compare_means(richness ~ Type.factor, group.by = c("Trt", "Time.Point"), BAlphaCombined, method = "wilcox")

```
#### Faith's phylogenetic diversity over time
```{r}
#####Statistical Test: Linear Model

#install.packages("mgcv")
library(mgcv)
#####Transform data so Character --> Categorical
nbphylo.diversity.join$Type <- as.factor(nbphylo.diversity.join$Type)
levels(nbphylo.diversity.join$Type)
nbphylo.diversity.join$Type

# Check summary statistics for each Type

BAlphaCombined$Type.factor <- ifelse(BAlphaCombined$Type %in% c("Spermosphere", "Rhizosphere"), "Both", "Bulk")

# Fit a GLM if needed
model_glm <- glm(richness ~ Type.factor*Time.Point*Trt, family = poisson(), data = BAlphaCombined)

# Summary of the GLM model
summary(model_glm)
anova(model_glm)
#install.packages("emmeans")
library("emmeans")

#install.packages("multcompView")
library(multcomp)
library(multcompView)

lsmeans <- emmeans(model_glm, ~Trt|Type.factor|Time.Point) # estimate lsmeans of variety within siteXyear 
Results_lsmeansEC <- multcomp::cld(lsmeans, alpha = 0.05, reversed = TRUE, details = TRUE, Letters = letters) # contrast with Tukey ajustment
Results_lsmeansEC
```

```{r}

bac.phylo.div <- BAlphaCombined %>%
  ggplot(aes(x = Type, y = PD, color = Steamed2)) +
  facet_wrap(~factor(Time, levels = c("Planting", "17hrs", "V2")), scale = "free") +
  stat_compare_means(method = "t.test") +
  geom_boxplot(position = position_dodge2(0.85, preserve = "single")) + 
    scale_y_continuous(limits = c(30, 210), breaks = seq(30, 210, 70)) +
  geom_point(position=position_jitterdodge(0.05)) +
  scale_fill_manual(values = cbbPalette ) +
  color_palette(cbbPalette) +
  #stat_summary(fun = mean,geom="line") +
  #stat_summary(fun.data = mean_se, geom = "errorbar", width = 0.5) +
  ylab("Faith's PD") +
  xlab("Type of Environment")+
  #geom_jitter(width = 0.1, alpha = 0.8)+
  #scale_color_manual(values=cbbPalette, name = "", labels = c("Planting", "17hrs", "V2")) +  
  #stat_compare_means(method = "kruskal", hide.ns = TRUE) +
  theme_classic()+
  theme(legend.text = element_text(face = "italic", size = 8), legend.title = element_blank())
bac.phylo.div

PDWil <- compare_means(PD ~ Trt, group.by = c("Type", "Time.Point"), BAlphaCombined, method = "wilcox")

summary(PDWil)

```
#### Bacterial evenness
```{r}
bac.even <- BAlphaCombined %>%
  ggplot(aes(x = Type, y = even, color = Steamed2)) +
  facet_wrap(~factor(Time, levels = c("Planting", "17hrs", "V2")), scale = "free") +
  scale_y_continuous(limits = c(0.45, 0.85), breaks = seq(0.45, 0.85, 0.05)) +
  stat_compare_means(method = "t.test") +
  geom_boxplot(position = position_dodge2(0.85, preserve = "single")) + 
  geom_point(position=position_jitterdodge(0.05)) +
  scale_fill_manual(values = cbbPalette ) +
  color_palette(cbbPalette) +
  #stat_summary(fun.y=mean,geom="bar", position = "dodge", width = 0.5) +
  #stat_summary(fun.data = mean_se, geom = "errorbar", width = 0.5) +
  ylab("Pielou's evenness") +
  xlab("Type of Environment")+
  #scale_color_manual(values=cbbPalette, name = "", labels = c("Planting", "17hrs", "V2")) +  
  theme_classic() +
  theme(legend.text = element_text(face = "italic", size = 8), legend.title = element_blank())
bac.even

EVWil <- compare_means(even ~ Type.factor, group.by = c("Trt", "Time.Point"), BAlphaCombined, method = "wilcox")

```
####Conjoined Figure
```{r}
figure1b <- ggpubr::ggarrange(bac.richness,
                                       bac.even,
                                       bac.phylo.div,
                                       labels = "auto",
                                       nrow = 3, ncol = 1, common.legend = T)
```
##Fungal Alpha Diversity##
###NonSteamed Fungal Alpha Diversity###
```{r}

fun.Nonsteamed@sam_data$shannon <- estimate_richness(fun.Nonsteamed, measures=c("Shannon"))$Shannon
fun.Nonsteamed@sam_data$invsimpson <- estimate_richness(fun.Nonsteamed, measures=c("InvSimpson"))$InvSimpson
fun.Nonsteamed@sam_data$simpson <- estimate_richness(fun.Nonsteamed, measures=c("Simpson"))$Simpson
fun.Nonsteamed@sam_data$richness <- estimate_richness(fun.Nonsteamed, measures=c("Observed"))$Observed
fun.Nonsteamed@sam_data$even <- fun.Nonsteamed@sam_data$shannon/log(fun.Nonsteamed@sam_data$richness)

Nsample.data.fun <- data.frame(fun.Nonsteamed@sam_data)

Nsample.data.fun$Time <- factor(Nsample.data.fun$Time, levels = c("17hrs", "Planting", "V2"))
```

```{r}
# Faith's phylogenetic diversity (I dont think we can do this with fungi)
fnonsteamed_otu <- fun.Nonsteamed@otu_table %>%
  as.data.frame()
nfphylo.diversity <- pd(t(fnonsteamed_otu), fun.Nonsteamed@phy_tree, include.root = F)
nfphylo.diversity$Samples <- rownames(nfphylo.diversity) 

nfphylo.diversity.join <- left_join(nfphylo.diversity, as.data.frame(fun.Nonsteamed@sam_data), by = c("Samples" = "Tube"))
nfphylo.diversity.join$Time.Point <- factor(nfphylo.diversity.join$Time, levels = c("17hrs", "Planting", "V2"))
####### Richness over time; supplemental figure part 1 #######
compare_means(richness ~ Trt, Sbphylo.diversity.join, group.by = "Time.Point")
compare_means(richness ~ Time, Ssample.data.bac)
```
####### Bacterial evenness; Figure 2a ########
```{r}
compare_means(even ~ Crop, sample.data.bac, group.by = "Time.Point")
```
###Steamed Fungal Alpha Diversity###
```{r}
fun.Steamed@sam_data$shannon <- estimate_richness(fun.Steamed, measures=c("Shannon"))$Shannon
fun.Steamed@sam_data$invsimpson <- estimate_richness(fun.Steamed, measures=c("InvSimpson"))$InvSimpson
fun.Steamed@sam_data$simpson <- estimate_richness(fun.Steamed, measures=c("Simpson"))$Simpson
fun.Steamed@sam_data$richness <- estimate_richness(fun.Steamed, measures=c("Observed"))$Observed
fun.Steamed@sam_data$even <- fun.Steamed@sam_data$shannon/log(fun.Steamed@sam_data$richness)

Ssample.data.fun <- data.frame(fun.Steamed@sam_data)

Ssample.data.fun$Time <- factor(Ssample.data.fun$Time, levels = c("17hrs", "Planting", "V2"))
```

###Conjoined Fungal Alpha Diversity###
```{r}

Ssample.data.fun$Steamed2 <- "Steamed"
Nsample.data.fun$Steamed2 <- "NonSteamed"

FAlphaCombined <- rbind.data.frame(Ssample.data.fun, Nsample.data.fun)

###RICHNESS###
compare_means(richness ~ Trt, Sbphylo.diversity.join, group.by = "Time.Point")
compare_means(richness ~ Time, Ssample.data.bac)


fun.richness <- FAlphaCombined %>%
  ggplot(aes(x = Type, y = richness, color = Steamed2)) +
  facet_wrap(~factor(Time, levels = c("Planting", "17hrs", "V2")), scale = "free") +
  stat_compare_means(method = "t.test") +
  geom_boxplot(position = position_dodge2(0.85, preserve = "single")) + 
  geom_point(position=position_jitterdodge(0.05)) +
  #stat_summary(fun = mean,geom="line") +
  #stat_summary(fun.data = mean_se, geom = "errorbar", width = 0.5) +
  ylab("Richness") +
  xlab("Type of Environment")+
  #geom_jitter(width = 0.1, alpha = 0.8)+
  #scale_color_manual(values=cbbPalette, name = "", labels = c("Planting", "17hrs", "V2")) +  
  #stat_compare_means(method = "kruskal", hide.ns = TRUE) +
  theme_classic()+
  theme(legend.text = element_text(face = "italic", size = 8), legend.title = element_blank())
fun.richness
```
####### Bacterial evenness; Figure 2a ########
```{r}
compare_means(even ~ Crop, sample.data.bac, group.by = "Time.Point")

fun.even <- FAlphaCombined %>%
  ggplot(aes(x = Type, y = even, color = Steamed2)) +
  facet_wrap(~factor(Time, levels = c("Planting", "17hrs", "V2")), scale = "free") +
  stat_compare_means(method = "t.test") +
  geom_boxplot(position = position_dodge2(0.85, preserve = "single")) + 
  geom_point(position=position_jitterdodge(0.05)) +
  #stat_summary(fun.y=mean,geom="bar", position = "dodge", width = 0.5) +
  #stat_summary(fun.data = mean_se, geom = "errorbar", width = 0.5) +
  ylab("Pielou's evenness") +
  xlab("Type of Environment")+
  #scale_color_manual(values=cbbPalette, name = "", labels = c("Planting", "17hrs", "V2")) +  
  theme_classic() +
  theme(legend.text = element_text(face = "italic", size = 8), legend.title = element_blank())
fun.even

```
#Beta Diversity#
###### Bacterial Beta-diversity using Bray ########
##### All Samples; Bray #####
```{r}
bac.bray = phyloseq::distance(BSpermosphere_RDS_CSS, "bray") # create Bray distance matrix (To switch with bray-curtis, change braycard to bray, and remove binary = true.)
```
### PERMANOVA - testing for differences in centroids
```{r}
set.seed(12345)
brayperm <- adonis2(bac.bray~Trt*Type*Time, as(sample_data(BSpermosphere_RDS_CSS), "data.frame"))

global.ord.bray <- phyloseq::ordinate(BSpermosphere_RDS_CSS, "MDS", "bray")
variation.axis1.bray <- round(100*global.ord.bray$values$Relative_eig[[1]], 2) # % variation on axis one
variation.axis2.bray <- round(100*global.ord.bray$values$Relative_eig[[2]], 2) # % variation on axis two

bac.ggplot.data.bray <- data.frame(global.ord.bray$vectors) # get the point values
bac.ggplot.data.bray$Tube <- rownames(bac.ggplot.data.bray) # add rownames
bac.ggplot.data.bray2 <- left_join(data.frame(cssbac.meta.data), bac.ggplot.data.bray,  by = "Tube") # THIS HAS TO BE THE SAME ON EACH FILE ##Purpose: add rest of metadata
bac.ggplot.data.bray2$Time.Point <- factor(bac.ggplot.data.bray2$Time, levels = c("Planting", "17hrs", "V2")) # order the time points in chronological order

bac.global.bray <- ggplot(bac.ggplot.data.bray2, aes(x = Axis.1, y = Axis.2, fill = Type, shape = Trt)) +
  facet_wrap(~factor(Time, levels = c("Planting", "17hrs", "V2")), scale = "free") +
  geom_point(size=4, alpha = 0.7)+
  scale_shape_manual(values=c(21,22, 23, 24,25,26,27), name = "Environment")+
  scale_fill_manual(values=cbbPalette, name = "Growth Stage")+
  xlab(paste("PcoA1 -", variation.axis1.bray, "%")) + 
  ylab(paste("PcoA2 -", variation.axis2.bray, "%")) +
  guides(fill=guide_legend(override.aes=list(shape= 21)),
         shape=guide_legend(override.aes=list(fill= "black")))+
  theme_classic() +
  ggtitle("Bray")
bac.global.bray

bac.global.bray2 <- ggplot(bac.ggplot.data.bray2, aes(x = Axis.1, y = Axis.2, fill = Trt, shape = Time)) +
  #facet_wrap(~factor(Time, levels = c("Planting", "17hrs", "V2")), scale = "free") +
  geom_point(size=4, alpha = 0.7)+
  scale_shape_manual(values=c(21,22, 23, 24,25,26,27), name = "Growth Stage")+
  scale_fill_manual(values=cbbPalette, name = "Environment")+
  xlab(paste("PcoA1 -", variation.axis1.bray, "%")) + 
  ylab(paste("PcoA2 -", variation.axis2.bray, "%")) +
  guides(fill=guide_legend(override.aes=list(shape= 21)),
         shape=guide_legend(override.aes=list(fill= "black")))+
  theme_classic() +
  ggtitle("Bray-Curtis")
bac.global.bray2

```
### 0 hours; Bray ####
```{r}
# filter to time point
bac_sperm_0 <- BSpermosphere_RDS_CSS %>% 
  subset_samples(Time == "Planting") %>% 
  phyloseq::filter_taxa(function(x) sum(x) > 0, TRUE)

# bray curtis distance
bac.bray.0 = phyloseq::distance(bac_sperm_0, "bray") # create bray distance matrix

# permanova
set.seed(12345)
adonis.output.0  <- adonis2(bac.bray.0~Trt*Type, as(sample_data(bac_sperm_0), "data.frame"))

pvalue0 <- adonis.output.0$`Pr(>F)`[[1]]
R2.0 <- adonis.output.0$R2[[1]]
```
###Dispersion for each Time-Point
```{r}
beta.disp.0 <- betadisper(bac.bray.0, bac_sperm_0@sam_data$Trt)

adonis.output.0
permutest(beta.disp.0)

sperm.0.ord <- phyloseq::ordinate(bac_sperm_0, "MDS", "bray")
sperm.0.ord.vectors <- data.frame(sperm.0.ord$vectors)
sperm.0.ord.vectors$Tube <- rownames(sperm.0.ord.vectors)
sperm.0.ord.vectors2 <- left_join(data.frame(bac_sperm_0@sam_data), sperm.0.ord.vectors,  by = "Tube")
sperm.0.ord.values <- data.frame(sperm.0.ord$values)
```
### percent variation
```{r}
variation.axis1.0.bray <- round(100*sperm.0.ord.values$Relative_eig[[1]], 2)
variation.axis2.0.bray <- round(100*sperm.0.ord.values$Relative_eig[[2]], 2)

sperm.0.ord.plot.bray <- ggplot(sperm.0.ord.vectors2, aes(x = Axis.1, y = Axis.2, fill = Trt, shape = Type)) +
  geom_point(size=4, alpha = 0.7) +
  scale_shape_manual(values = c(21,24)) +
  scale_fill_manual(values=cbbPalette) + 
  xlab(paste("PcoA1 -", variation.axis1.0.bray, "%")) + 
  ylab(paste("PcoA2 -", variation.axis1.0.bray, "%")) +
  ggtitle(paste("Planting")) +
  theme_bw() +
  theme(legend.position="bottom",
        legend.title=element_blank(),
        legend.key = element_blank(),
        plot.title=element_text(size=10, face = "bold")) 
sperm.0.ord.plot.bray
```
### 17 hours; Bray ####
### filter to time point
```{r}
bac_sperm_17 <- BSpermosphere_RDS_CSS %>% 
  subset_samples(Time == "17hrs") %>% 
  phyloseq::filter_taxa(function(x) sum(x) > 0, TRUE)

# bray curtis distance
bac.bray.17 = phyloseq::distance(bac_sperm_17, "bray") # create Bray distance matrix

# permanova
set.seed(12345)
adonis.output.17  <- adonis2(bac.bray.17~Trt*Type, as(sample_data(bac_sperm_17), "data.frame"))

pvalue.17.bray <- adonis.output.17$`Pr(>F)`[[1]]
R2.17.bray <- adonis.output.17$R2[[1]]
```
###Dispersion for each Time-Point
```{r}
beta.disp.17.bray <- betadisper(bac.bray.17, bac_sperm_17@sam_data$Trt)

adonis.output.17
permutest(beta.disp.17.bray)

sperm.17.ord.bray <- phyloseq::ordinate(bac_sperm_17, "MDS", "bray")
sperm.17.ord.vectors.bray <- data.frame(sperm.17.ord.bray$vectors)
sperm.17.ord.vectors.bray$Tube <- rownames(sperm.17.ord.vectors.bray)
sperm.17.ord.vectors2.bray <- left_join(data.frame(bac_sperm_17@sam_data), sperm.17.ord.vectors.bray,  by = "Tube")
sperm.17.ord.values.bray <- data.frame(sperm.17.ord.bray$values)
```
### percent variation
```{r}
variation.axis1.17.bray <- round(166*sperm.17.ord.values.bray$Relative_eig[[1]], 2)
variation.axis2.17.bray <- round(166*sperm.17.ord.values.bray$Relative_eig[[2]], 2)

sperm.17.ord.plot.bray <- ggplot(sperm.17.ord.vectors2.bray, aes(x = Axis.1, y = Axis.2, fill = Trt, shape = Type)) +
  geom_point(size=4, alpha = 0.7) +
  scale_shape_manual(values = c(21,24)) +
  scale_fill_manual(values=cbbPalette) +
  xlab(paste("PcoA1 -", variation.axis1.17.bray, "%")) + 
  ylab(paste("PcoA2 -", variation.axis2.17.bray, "%")) +
  guides(fill=guide_legend(override.aes=list(shape=21))) +
  ggtitle(paste("17 Hours Postplanting")) +
  theme_bw() +
  theme(legend.position="bottom",
        legend.title=element_blank(),
        legend.key = element_blank(),
        plot.title=element_text(size=10, face = "bold"))
sperm.17.ord.plot.bray
```
### V2 hours; bray ####
### filter to time point
```{r}
bac_sperm_V2 <- BSpermosphere_RDS_CSS %>% 
  subset_samples(Time == "V2") %>% 
  phyloseq::filter_taxa(function(x) sum(x) > 0, TRUE)

# bray curtis distance
bac.bray.V2 = phyloseq::distance(bac_sperm_V2, "bray") # create bray distance matrix

# permanova
set.seed(12345)
adonis.output.V2  <- adonis2(bac.bray.V2~Trt*Type, as(sample_data(bac_sperm_V2), "data.frame"))

pvalue.V2 <- adonis.output.V2$`Pr(>F)`[[1]]
R2.V2 <- adonis.output.V2$R2[[1]]
```
###Dispersion for each Time-Point
```{r}
beta.disp.V2 <- betadisper(bac.bray.V2, bac_sperm_V2@sam_data$Trt)

adonis.output.V2
permutest(beta.disp.V2)

sperm.V2.ord <- phyloseq::ordinate(bac_sperm_V2, "MDS", "bray")
sperm.V2.ord.vectors <- data.frame(sperm.V2.ord$vectors)
sperm.V2.ord.vectors$Tube <- rownames(sperm.V2.ord.vectors)
sperm.V2.ord.vectors2 <- left_join(data.frame(bac_sperm_V2@sam_data), sperm.V2.ord.vectors,  by = "Tube")
sperm.V2.ord.values <- data.frame(sperm.V2.ord$values)
```
### percent variation
```{r}
variation.axis1.V2.bray <- round(100*sperm.V2.ord.values$Relative_eig[[1]], 2)
variation.axis2.V2.bray <- round(100*sperm.V2.ord.values$Relative_eig[[2]], 2)

sperm.V2.ord.plot.bray <- ggplot(sperm.V2.ord.vectors2, aes(x = Axis.1, y = Axis.2, fill = Trt, shape = Type)) +
  geom_point(size=4, alpha = 0.7) +
  scale_shape_manual(values = c(21,25)) +
  scale_fill_manual(values=cbbPalette) +
  xlab(paste("PcoA1 -", variation.axis1.V2.bray, "%")) + 
  ylab(paste("PcoA2 -", variation.axis2.V2.bray, "%")) +
  theme(legend.position="bottom",legend.title=element_blank(),legend.key = element_blank()) +
  guides(fill=guide_legend(override.aes=list(shape=21))) +
  ggtitle(paste("V2 Growth Stage")) +
  theme_bw() +
  theme(legend.position="bottom",
        legend.title=element_blank(),
        legend.key = element_blank(),
        plot.title=element_text(size=10, face = "bold"))
sperm.V2.ord.plot.bray
pcoa_bac.bray <- ggpubr::ggarrange(sperm.0.ord.plot.bray,
                               sperm.17.ord.plot.bray,
                               sperm.V2.ord.plot.bray,
                               labels = "auto",
                               nrow = 1, ncol = 3, common.legend = TRUE, legend = "bottom")
pcoa_bac.bray

#Manually change the legend on power-point
#yellow circles are steamed bulk soil
#Show PCOA plots in equal proportions in respects with the axis
#Manually put in P-Value cause coding it is not worth it
###
#Bray takes into account Presence absence and its abundance
#Jacaard is only presence / absence
#unifrac Weighted = weighted by abundancies, unweighted = raw P/A
```
###### Beta-diversity using Jaccard ########
##### All Samples; jaccard #####
```{r}
bac.jac = phyloseq::distance(BSpermosphere_RDS_CSS, "jaccard", ) # create Jaccard distance matrix (To switch with bray-curtis, change jaccard to bray, and remove .)
```
##Bacterial Beta Diversity
### PERMANOVA - testing for differences in centroids
```{r}
set.seed(12345)
adonis2(bac.jac~Trt*Time, as(sample_data(BSpermosphere_RDS_CSS), "data.frame"))

global.ord.jac <- phyloseq::ordinate(BSpermosphere_RDS_CSS, "MDS", "jaccard", Binary = TRUE)
variation.axis1.jac <- round(100*global.ord.jac$values$Relative_eig[[1]], 2) # % variation on axis one
variation.axis2.jac <- round(100*global.ord.jac$values$Relative_eig[[2]], 2) # % variation on axis two

bac.ggplot.data.jac <- data.frame(global.ord.jac$vectors) # get the point values
bac.ggplot.data.jac$Tube <- rownames(bac.ggplot.data.jac) # add rownames
bac.ggplot.data.jac2 <- left_join(data.frame(cssbac.meta.data), bac.ggplot.data.jac,  by = "Tube") # THIS HAS TO BE THE SAME ON EACH FILE ##Purpose: add rest of metadata
bac.ggplot.data.jac2$Time.Point <- factor(bac.ggplot.data.jac2$Time, levels = c("Planting", "17hrs", "V2")) # order the time points in chronological order

bac.global.jac <- ggplot(bac.ggplot.data.jac2, aes(x = Axis.1, y = Axis.2, fill = Type, shape = Trt)) +
  facet_wrap(~factor(Time, levels = c("Planting", "17hrs", "V2")), scale = "free") +
  geom_point(size=4, alpha = 0.7)+
  scale_shape_manual(values=c(21,22, 23, 24,25,26,27), name = "Environment")+
  scale_fill_manual(values=cbbPalette, name = "Growth Stage")+
  xlab(paste("PcoA1 -", variation.axis1.jac, "%")) + 
  ylab(paste("PcoA2 -", variation.axis2.jac, "%")) +
  guides(fill=guide_legend(override.aes=list(shape= 21)),
         shape=guide_legend(override.aes=list(fill= "black")))+
  theme_classic() +
  ggtitle("Jaccards Analysis")
bac.global.jac

bac.global.jac2 <- ggplot(bac.ggplot.data.jac2, aes(x = Axis.1, y = Axis.2, fill = Trt, shape = Time)) +
  #facet_wrap(~factor(Time, levels = c("Planting", "17hrs", "V2")), scale = "free") +
  geom_point(size=4, alpha = 0.7)+
  scale_shape_manual(values=c(21,22, 23, 24,25,26,27), name = "Growth Stage")+
  scale_fill_manual(values=cbbPalette, name = "Environment")+
  xlab(paste("PcoA1 -", variation.axis1.jac, "%")) + 
  ylab(paste("PcoA2 -", variation.axis2.jac, "%")) +
  guides(fill=guide_legend(override.aes=list(shape= 21)),
         shape=guide_legend(override.aes=list(fill= "black")))+
  theme_classic() +
  ggtitle("Bray-Curtis")
bac.global.jac2

```
### Separate analysis based on time-points #
```{r}
###THIS SECTION ONLY MATTERS IF 2 TIMEPOINTS ARE VERY SIMILAR
#### 0 v. 17 hours
bac_sperm_0vs17 <- BSpermosphere_RDS_CSS %>% 
  subset_samples(Time %in% c("Planting", "17hrs", "V2")) %>% 
  phyloseq::filter_taxa(function(x) sum(x) > 0, TRUE)

# bray curtis distance
bac.bray.0vs17 = phyloseq::distance(bac_sperm_0vs17, "bray") # create bray-curtis distance matrix

# permanova
set.seed(12325)
adonis.output.0vs17 <- adonis2(bac.bray.0vs17~Trt*Time, as(sample_data(bac_sperm_0vs17), "data.frame"))
###adonis result there is a difference in Trt vs Time
adonis.output.0vs17.2 <- adonis2(bac.bray.0vs17~Type*Time, as(sample_data(bac_sperm_0vs17), "data.frame"))
###There is no relationship of Type x Time
adonis.output.0vs17.3 <- adonis2(bac.bray.0vs17~Type*Trt, as(sample_data(bac_sperm_0vs17), "data.frame"))
###There is a difference in Type x Trt (Sanity Check)

```
### 0 hours; Jaccard ####
```{r}
# filter to time point
bac_sperm_0 <- BSpermosphere_RDS_CSS %>% 
  subset_samples(Time == "Planting") %>% 
  phyloseq::filter_taxa(function(x) sum(x) > 0, TRUE)

# bray curtis distance
bac.jac.0 = phyloseq::distance(bac_sperm_0, "jaccard", binary = TRUE) # create Jaccard distance matrix

# permanova
set.seed(12345)
adonis.output.0  <- adonis2(bac.jac.0~Trt*Type, as(sample_data(bac_sperm_0), "data.frame"))

pvalue0 <- adonis.output.0$`Pr(>F)`[[1]]
R2.0 <- adonis.output.0$R2[[1]]
```
###Dispersion for each Time-Point
```{r}
beta.disp.0 <- betadisper(bac.jac.0, bac_sperm_0@sam_data$Trt)

adonis.output.0
permutest(beta.disp.0)

sperm.0.ord <- phyloseq::ordinate(bac_sperm_0, "MDS", "jaccard", binary = TRUE)
sperm.0.ord.vectors <- data.frame(sperm.0.ord$vectors)
sperm.0.ord.vectors$Tube <- rownames(sperm.0.ord.vectors)
sperm.0.ord.vectors2 <- left_join(data.frame(bac_sperm_0@sam_data), sperm.0.ord.vectors,  by = "Tube")
sperm.0.ord.values <- data.frame(sperm.0.ord$values)
```
### percent variation
```{r}
variation.axis1.0.jac <- round(100*sperm.0.ord.values$Relative_eig[[1]], 2)
variation.axis2.0.jac <- round(100*sperm.0.ord.values$Relative_eig[[2]], 2)

sperm.0.ord.plot <- ggplot(sperm.0.ord.vectors2, aes(x = Axis.1, y = Axis.2, fill = Trt, shape = Type)) +
  geom_point(size=4, alpha = 0.7) +
  scale_shape_manual(values = c(21,24)) +
  scale_fill_manual(values=cbbPalette) + 
  xlab(paste("PcoA1 -", variation.axis1.0.jac, "%")) + 
  ylab(paste("PcoA2 -", variation.axis1.0.jac, "%")) +
  ggtitle(paste("0 hrs")) +
  theme_bw() +
  theme(legend.position="bottom",
        legend.title=element_blank(),
        legend.key = element_blank(),
        plot.title=element_text(size=10, face = "bold")) 
sperm.0.ord.plot
```
### 17 hours; Jaccard ####
### filter to time point
```{r}
bac_sperm_17 <- BSpermosphere_RDS_CSS %>% 
  subset_samples(Time == "17hrs") %>% 
  phyloseq::filter_taxa(function(x) sum(x) > 0, TRUE)

# bray curtis distance
bac.jac.17 = phyloseq::distance(bac_sperm_17, "jaccard", binary = TRUE) # create Jaccard distance matrix

# permanova
set.seed(12345)
adonis.output.17  <- adonis2(bac.jac.17~Trt*Type, as(sample_data(bac_sperm_17), "data.frame"))

pvalue.17 <- adonis.output.17$`Pr(>F)`[[1]]
R2.17 <- adonis.output.17$R2[[1]]
```
###Dispersion for each Time-Point
```{r}
beta.disp.17 <- betadisper(bac.jac.17, bac_sperm_17@sam_data$Trt)

adonis.output.17
permutest(beta.disp.17)

sperm.17.ord <- phyloseq::ordinate(bac_sperm_17, "MDS", "jaccard", binary = TRUE)
sperm.17.ord.vectors <- data.frame(sperm.17.ord$vectors)
sperm.17.ord.vectors$Tube <- rownames(sperm.17.ord.vectors)
sperm.17.ord.vectors2 <- left_join(data.frame(bac_sperm_17@sam_data), sperm.17.ord.vectors,  by = "Tube")
sperm.17.ord.values <- data.frame(sperm.17.ord$values)
```
### percent variation
```{r}
variation.axis1.17.jac <- round(166*sperm.17.ord.values$Relative_eig[[1]], 2)
variation.axis2.17.jac <- round(166*sperm.17.ord.values$Relative_eig[[2]], 2)

sperm.17.ord.plot <- ggplot(sperm.17.ord.vectors2, aes(x = Axis.1, y = Axis.2, fill = Trt, shape = Type)) +
  geom_point(size=4, alpha = 0.7) +
  scale_shape_manual(values = c(21,24)) +
  scale_fill_manual(values=cbbPalette) +
  xlab(paste("PcoA1 -", variation.axis1.17.jac, "%")) + 
  ylab(paste("PcoA2 -", variation.axis2.17.jac, "%")) +
  guides(fill=guide_legend(override.aes=list(shape=21))) +
  ggtitle(paste("17 hrs")) +
  theme_bw() +
  theme(legend.position="bottom",
        legend.title=element_blank(),
        legend.key = element_blank(),
        plot.title=element_text(size=10, face = "bold"))
sperm.17.ord.plot
```
### V2 hours; jaccard ####
### filter to time point
```{r}
bac_sperm_V2 <- BSpermosphere_RDS_CSS %>% 
  subset_samples(Time == "V2") %>% 
  phyloseq::filter_taxa(function(x) sum(x) > 0, TRUE)

# bray curtis distance
bac.jac.V2 = phyloseq::distance(bac_sperm_V2, "jaccard", binary = TRUE) # create Jaccard distance matrix

# permanova
set.seed(12345)
adonis.output.V2  <- adonis2(bac.jac.V2~Trt*Type, as(sample_data(bac_sperm_V2), "data.frame"))

pvalue.V2 <- adonis.output.V2$`Pr(>F)`[[1]]
R2.V2 <- adonis.output.V2$R2[[1]]
```
###Dispersion for each Time-Point
```{r}
beta.disp.V2 <- betadisper(bac.jac.V2, bac_sperm_V2@sam_data$Trt)

adonis.output.V2
permutest(beta.disp.V2)

sperm.V2.ord <- phyloseq::ordinate(bac_sperm_V2, "MDS", "jaccard", binary = TRUE)
sperm.V2.ord.vectors <- data.frame(sperm.V2.ord$vectors)
sperm.V2.ord.vectors$Tube <- rownames(sperm.V2.ord.vectors)
sperm.V2.ord.vectors2 <- left_join(data.frame(bac_sperm_V2@sam_data), sperm.V2.ord.vectors,  by = "Tube")
sperm.V2.ord.values <- data.frame(sperm.V2.ord$values)
```
### percent variation
```{r}
variation.axis1.V2.jac <- round(100*sperm.V2.ord.values$Relative_eig[[1]], 2)
variation.axis2.V2.jac <- round(100*sperm.V2.ord.values$Relative_eig[[2]], 2)

sperm.V2.ord.plot <- ggplot(sperm.V2.ord.vectors2, aes(x = Axis.1, y = Axis.2, fill = Trt, shape = Type)) +
  geom_point(size=4, alpha = 0.7) +
  scale_shape_manual(values = c(21,25)) +
  scale_fill_manual(values=cbbPalette) +
  xlab(paste("PcoA1 -", variation.axis1.V2.jac, "%")) + 
  ylab(paste("PcoA2 -", variation.axis2.V2.jac, "%")) +
  theme(legend.position="bottom",legend.title=element_blank(),legend.key = element_blank()) +
  guides(fill=guide_legend(override.aes=list(shape=21))) +
  ggtitle(paste("V2 Growth Stage")) +
  theme_bw() +
  theme(legend.position="bottom",
        legend.title=element_blank(),
        legend.key = element_blank(),
        plot.title=element_text(size=10, face = "bold"))
sperm.V2.ord.plot
pcoa_bac.jac <- ggpubr::ggarrange(sperm.0.ord.plot,
                               sperm.17.ord.plot,
                               sperm.V2.ord.plot,
                               labels = "auto",
                               nrow = 1, ncol = 3, common.legend = TRUE, legend = "bottom")
pcoa_bac.jac

#Manually change the legend on power-point
#yellow circles are steamed bulk soil
#Show PCOA plots in equal proportions in respects with the axis
#Manually put in P-Value cause coding it is not worth it
###
#Bray takes into account Presence absence and its abundance
#Jacaard is only presence / absence
#unifrac Weighted = weighted by abundancies, unweighted = raw P/A
```
###### Beta-Diversity using Weighted Unifrac Only Bacteria ####
```{r}
##### All Samples ########
bac.unifrac = UniFrac(BSpermosphere_RDS_CSS, weighted = TRUE) # create weighted unifrac distance matrix

# PERMANOVA - testing for differences in centroids
set.seed(12345)
adonis2(bac.unifrac~Trt*Type*Time, as(sample_data(BSpermosphere_RDS_CSS), "data.frame"))

global.ord.uni <- phyloseq::ordinate(BSpermosphere_RDS_CSS, "MDS", "unifrac", weighted = TRUE)
variation.axis1 <- round(100*global.ord.uni$values$Relative_eig[[1]], 2)
variation.axis2 <- round(100*global.ord.uni$values$Relative_eig[[2]], 2)

bac.ggplot.data.uni <- data.frame(global.ord.uni$vectors)
bac.ggplot.data.uni$Tube <- rownames(bac.ggplot.data.uni)
bac.ggplot.data.uni2 <- left_join(data.frame(BSpermosphere_RDS_CSS@sam_data), bac.ggplot.data.uni,  by = "Tube")
bac.ggplot.data.uni2$Time.Point <- factor(bac.ggplot.data.uni2$Time, levels = c("Planting", "17hrs", "V2"))

global.uni <- ggplot(bac.ggplot.data.uni2, aes(x = Axis.1, y = Axis.2, fill = Trt, shape = Type)) +
  geom_point(size=4, alpha = 0.7)+
  scale_shape_manual(values=c(21,22,24,25))+
  scale_fill_manual(values= cbbPalette)+
  xlab(paste("PcoA1 -", variation.axis1, "%")) + 
  ylab(paste("PcoA2 -", variation.axis2, "%")) + theme(legend.position="bottom",
        legend.title=element_blank(),
        legend.key = element_blank(),
        plot.title=element_text(size=10, face = "bold")) +
  #guides(fill=guide_legend(override.aes=list(shape= 21)),
         #shape=guide_legend(override.aes=list(fill= "black")))+ 
  ggtitle("Weighted Unifrac")

global.uni

### Supplemental Figure 2
Supplemental.figure2 <- ggpubr::ggarrange(global.uni, global.bray, common.legend = TRUE, labels = c("a", "b"), legend = "right")
```
### 0 hours; Unifrac #####
```{r}
bac_sperm_0 <- BSpermosphere_RDS_CSS %>% 
  subset_samples(Time == "Planting") %>% 
  phyloseq::filter_taxa(function(x) sum(x) > 0, TRUE)

# PERMANOVA - testing for differences in centroids
set.seed(12345)
prok.dist.uni.0hrs = UniFrac(bac_sperm_0, weighted = T) # create distance matrix
adonis.0.uni <- adonis2(prok.dist.uni.0hrs~Trt*Type, as(sample_data(bac_sperm_0), "data.frame")) #Are there significant changes?
adonis.0.uni
# The R2 is the variation due to different factors 
# pvalue is the pvalue for the factor

pvalue <- adonis.0.uni$`Pr(>F)`[[1]]
R2 <- adonis.0.uni$R2[[1]]

# ploting
ordination.unifrac.pcoa.0hrs <- ordinate(bac_sperm_0, "PCoA", distance = prok.dist.uni.0hrs) # calculate the resemblance and ordinate using PCoA
uni.0.values <- data.frame(ordination.unifrac.pcoa.0hrs$values)

# percent variation
variation.axis1.0.uni <- round(100*uni.0.values$Relative_eig[[1]], 2)
variation.axis2.0.uni <- round(100*uni.0.values$Relative_eig[[2]], 2)

# data from ordinations
uni.0.vectors <- data.frame(ordination.unifrac.pcoa.0hrs$vectors) # positions of your points on the pcoa graph
uni.0.vectors$Tube <- rownames(uni.0.vectors)
uni.0.vectors2 <- left_join(data.frame(bac_sperm_0@sam_data), uni.0.vectors,  by = "Tube")

 pcoa.unifrac.0hrs.new <- ggplot(uni.0.vectors2, aes(x = Axis.1, y = Axis.2, fill = Trt, shape = Type)) +
  geom_point(size=4, alpha = 0.7) +
  scale_shape_manual(values = c(21,25)) +
  scale_fill_manual(values=cbbPalette) +
  xlab(paste("PcoA1 -", variation.axis1.0.uni, "%")) + 
  ylab(paste("PcoA2 -", variation.axis2.0.uni, "%")) +
  guides(fill=guide_legend(override.aes=list(shape=21))) +
  ggtitle(paste("Planting")) +
  theme_bw() +
  theme(legend.position="bottom",
        legend.title=element_blank(),
        legend.key = element_blank(),
        plot.title=element_text(size=10, face = "bold"))
pcoa.unifrac.0hrs.new

#Dispersion for each Time-Point
beta.disp.0.uni <- betadisper(prok.dist.uni.0hrs, bac_sperm_0@sam_data$Trt)
permutest(beta.disp.0.uni)
```
### 17 hours; Unifrac #####
```{r}
bac_sperm_17 <- BSpermosphere_RDS_CSS %>% 
  subset_samples(Time == "17hrs") %>% 
  phyloseq::filter_taxa(function(x) sum(x) > 0, TRUE)

# PERMANOVA - testing for differences in centroids
set.seed(12345)
prok.dist.uni.17hrs = UniFrac(bac_sperm_17, weighted = T) # create distance matrix
adonis.17.uni <- adonis2(prok.dist.uni.17hrs~Trt*Type, as(sample_data(bac_sperm_17), "data.frame")) #Are there significant changes?
adonis.17.uni
# The R2 is the variation due to different factors 
# pvalue is the pvalue for the factor

pvalue <- adonis.17.uni$`Pr(>F)`[[1]]
R2 <- adonis.17.uni$R2[[1]]

# ploting
ordination.unifrac.pcoa.17hrs <- ordinate(bac_sperm_17, "PCoA", distance = prok.dist.uni.17hrs) # calculate the resemblance and ordinate using PCoA
uni.17.values <- data.frame(ordination.unifrac.pcoa.17hrs$values)

# percent variation
variation.axis1.17.uni <- round(100*uni.17.values$Relative_eig[[1]], 2)
variation.axis2.17.uni <- round(100*uni.17.values$Relative_eig[[2]], 2)

# data from ordinations
uni.17.vectors <- data.frame(ordination.unifrac.pcoa.17hrs$vectors) # positions of your points on the pcoa graph
uni.17.vectors$Tube <- rownames(uni.17.vectors)
uni.17.vectors2 <- left_join(data.frame(bac_sperm_17@sam_data), uni.17.vectors,  by = "Tube")

  pcoa.unifrac.17hrs.new <- ggplot(uni.17.vectors2, aes(x = Axis.1, y = Axis.2, fill = Trt, shape = Type)) +
  geom_point(size=4, alpha = 0.7) +
  scale_shape_manual(values = c(21,25)) +
  scale_fill_manual(values=cbbPalette) +
  xlab(paste("PcoA1 -", variation.axis1.17.uni, "%")) + 
  ylab(paste("PcoA2 -", variation.axis2.17.uni, "%")) +
  guides(fill=guide_legend(override.aes=list(shape=21))) +
  ggtitle(paste("17 Hours Postplanting")) +
  theme_bw() +
  theme(legend.position="bottom",
        legend.title=element_blank(),
        legend.key = element_blank(),
        plot.title=element_text(size=10, face = "bold"))
pcoa.unifrac.17hrs.new

#Dispersion for each Time-Point
beta.disp.17.uni <- betadisper(prok.dist.uni.17hrs, bac_sperm_17@sam_data$Trt)
permutest(beta.disp.17.uni)
```
### V2; Unifrac #####
```{r}
bac_sperm_V2 <- BSpermosphere_RDS_CSS %>% 
  subset_samples(Time == "V2") %>% 
  phyloseq::filter_taxa(function(x) sum(x) > 0, TRUE)

# PERMANOVA - testing for differences in centroids
set.seed(12345)
prok.dist.uni.V2hrs = UniFrac(bac_sperm_V2, weighted = T) # create distance matrix
adonis.V2.uni <- adonis2(prok.dist.uni.V2hrs~Trt*Type, as(sample_data(bac_sperm_V2), "data.frame")) #Are there significant changes?
adonis.V2.uni
# The R2 is the variation due to different factors 
# pvalue is the pvalue for the factor

pvalue <- adonis.V2.uni$`Pr(>F)`[[1]]
R2 <- adonis.V2.uni$R2[[1]]

# ploting
ordination.unifrac.pcoa.V2hrs <- ordinate(bac_sperm_V2, "PCoA", distance = prok.dist.uni.V2hrs) # calculate the resemblance and ordinate using PCoA
uni.V2.values <- data.frame(ordination.unifrac.pcoa.V2hrs$values)

# percent variation
variation.axis1.V2.uni <- round(100*uni.V2.values$Relative_eig[[1]], 2)
variation.axis2.V2.uni <- round(100*uni.V2.values$Relative_eig[[2]], 2)

# data from ordinations
uni.V2.vectors <- data.frame(ordination.unifrac.pcoa.V2hrs$vectors) # positions of your points on the pcoa graph
uni.V2.vectors$Tube <- rownames(uni.V2.vectors)
uni.V2.vectors2 <- left_join(data.frame(bac_sperm_V2@sam_data), uni.V2.vectors,  by = "Tube")

pcoa.unifrac.V2hrs.new <- ggplot(uni.V2.vectors2, aes(x = Axis.1, y = Axis.2, fill = Trt, shape = Type)) +
  geom_point(size=4, alpha = 0.7) +
  scale_shape_manual(values = c(21,25)) +
  scale_fill_manual(values=cbbPalette) +
  xlab(paste("PcoA1 -", variation.axis1.V2.uni, "%")) + 
  ylab(paste("PcoA2 -", variation.axis2.V2.uni, "%")) +
  guides(fill=guide_legend(override.aes=list(shape=21))) +
  ggtitle(paste("V2 Growth Stage")) +
  theme_bw() +
  theme(legend.position="bottom",
        legend.title=element_blank(),
        legend.key = element_blank(),
        plot.title=element_text(size=10, face = "bold"))
pcoa.unifrac.V2hrs.new



#Dispersion for each Time-Point
beta.disp.V2.uni <- betadisper(prok.dist.uni.V2hrs, bac_sperm_V2@sam_data$Trt)
permutest(beta.disp.V2.uni)
```
##### Figure 1; Stick all together #####
```{r}
figure1 <- ggpubr::ggarrange(pcoa.unifrac.0hrs.new, 
                                       pcoa.unifrac.17hrs.new,
                                       pcoa.unifrac.V2hrs.new,
                                       sperm.0.ord.plot, 
                                       sperm.17.ord.plot,
                                       sperm.V2.ord.plot,
                                       labels = "auto",
                                       nrow = 2, ncol = 3, common.legend = T)
```

###### Beta-Diversity using UnWeighted Unifrac Only Bacteria ####
```{r}
##### All Samples ########
bac.unifrac = UniFrac(BSpermosphere_RDS_CSS, weighted = FALSE) # create weighted unifrac distance matrix

# PERMANOVA - testing for differences in centroids
set.seed(12345)
adonis2(bac.unifrac~Trt*Type*Time, as(sample_data(BSpermosphere_RDS_CSS), "data.frame"))

global.ord.uni <- phyloseq::ordinate(BSpermosphere_RDS_CSS, "MDS", "unifrac", weighted = FALSE)
variation.axis1 <- round(100*global.ord.uni$values$Relative_eig[[1]], 2)
variation.axis2 <- round(100*global.ord.uni$values$Relative_eig[[2]], 2)

bac.ggplot.data.uni <- data.frame(global.ord.uni$vectors)
bac.ggplot.data.uni$Tube <- rownames(bac.ggplot.data.uni)
bac.ggplot.data.uni2 <- left_join(data.frame(BSpermosphere_RDS_CSS@sam_data), bac.ggplot.data.uni,  by = "Tube")
bac.ggplot.data.uni2$Time.Point <- factor(bac.ggplot.data.uni2$Time, levels = c("Planting", "17hrs", "V2"))

global.uni.unweighted <- ggplot(bac.ggplot.data.uni2, aes(x = Axis.1, y = Axis.2, fill = Trt, shape = Type)) +
  geom_point(size=4, alpha = 0.7)+
  scale_shape_manual(values=c(21,22,24,25))+
  scale_fill_manual(values= cbbPalette)+
  xlab(paste("PcoA1 -", variation.axis1, "%")) + 
  ylab(paste("PcoA2 -", variation.axis2, "%")) +
  #guides(fill=guide_legend(override.aes=list(shape= 21)),
         #shape=guide_legend(override.aes=list(fill= "black")))+ 
  ggtitle("Weighted Unifrac")

global.uni.unweighted

### Supplemental Figure 2
Supplemental.figure2 <- ggpubr::ggarrange(global.uni, global.bray, common.legend = TRUE, labels = c("a", "b"), legend = "right")
```
### 0 hours; Unweighted Unifrac #####
```{r}
bac_sperm_0 <- BSpermosphere_RDS_CSS %>% 
  subset_samples(Time == "Planting") %>% 
  phyloseq::filter_taxa(function(x) sum(x) > 0, TRUE)

# PERMANOVA - testing for differences in centroids
set.seed(12345)
prok.dist.uni.0hrs = UniFrac(bac_sperm_0, weighted = FALSE) # create distance matrix
adonis.0.uni <- adonis2(prok.dist.uni.0hrs~Trt*Type, as(sample_data(bac_sperm_0), "data.frame")) #Are there significant changes?
adonis.0.uni
# The R2 is the variation due to different factors 
# pvalue is the pvalue for the factor

pvalue <- adonis.0.uni$`Pr(>F)`[[1]]
R2 <- adonis.0.uni$R2[[1]]

# ploting
ordination.unifrac.pcoa.0hrs <- ordinate(bac_sperm_0, "PCoA", distance = prok.dist.uni.0hrs) # calculate the resemblance and ordinate using PCoA
uni.0.values <- data.frame(ordination.unifrac.pcoa.0hrs$values)

# percent variation
variation.axis1.0.uni <- round(100*uni.0.values$Relative_eig[[1]], 2)
variation.axis2.0.uni <- round(100*uni.0.values$Relative_eig[[2]], 2)

# data from ordinations
uni.0.vectors <- data.frame(ordination.unifrac.pcoa.0hrs$vectors) # positions of your points on the pcoa graph
uni.0.vectors$Tube <- rownames(uni.0.vectors)
uni.0.vectors2 <- left_join(data.frame(bac_sperm_0@sam_data), uni.0.vectors,  by = "Tube")

un.pcoa.unifrac.0hrs.new <- ggplot(uni.0.vectors2, aes(x = Axis.1, y = Axis.2, fill = Trt, shape = Type)) +
  geom_point(size=4, alpha = 0.7) +
  scale_shape_manual(values = c(21,25)) +
  scale_fill_manual(values=cbbPalette) +
  xlab(paste("PcoA1 -", variation.axis1.0.uni, "%")) + 
  ylab(paste("PcoA2 -", variation.axis2.0.uni, "%")) +
  guides(fill=guide_legend(override.aes=list(shape=21))) +
  ggtitle(paste("Planting")) +
  theme_bw() +
  theme(legend.position="bottom",
        legend.title=element_blank(),
        legend.key = element_blank(),
        plot.title=element_text(size=10, face = "bold"))
un.pcoa.unifrac.0hrs.new

#Dispersion for each Time-Point
beta.disp.0.uni <- betadisper(prok.dist.uni.0hrs, bac_sperm_0@sam_data$Trt)
permutest(beta.disp.0.uni)
beta.disp.0.uni.Ty <- betadisper(prok.dist.uni.0hrs, bac_sperm_0@sam_data$Type)

```
### 17 hours; Unweighted Unifrac #####
```{r}
bac_sperm_17 <- BSpermosphere_RDS_CSS %>% 
  subset_samples(Time == "17hrs") %>% 
  phyloseq::filter_taxa(function(x) sum(x) > 0, TRUE)

# PERMANOVA - testing for differences in centroids
set.seed(12345)
prok.dist.uni.17hrs = UniFrac(bac_sperm_17, weighted = F) # create distance matrix
adonis.17.uni <- adonis2(prok.dist.uni.17hrs~Trt*Type, as(sample_data(bac_sperm_17), "data.frame")) #Are there significant changes?
adonis.17.uni
# The R2 is the variation due to different factors 
# pvalue is the pvalue for the factor

pvalue <- adonis.17.uni$`Pr(>F)`[[1]]
R2 <- adonis.17.uni$R2[[1]]

# ploting
ordination.unifrac.pcoa.17hrs <- ordinate(bac_sperm_17, "PCoA", distance = prok.dist.uni.17hrs) # calculate the resemblance and ordinate using PCoA
uni.17.values <- data.frame(ordination.unifrac.pcoa.17hrs$values)

# percent variation
variation.axis1.17.uni <- round(100*uni.17.values$Relative_eig[[1]], 2)
variation.axis2.17.uni <- round(100*uni.17.values$Relative_eig[[2]], 2)

# data from ordinations
uni.17.vectors <- data.frame(ordination.unifrac.pcoa.17hrs$vectors) # positions of your points on the pcoa graph
uni.17.vectors$Tube <- rownames(uni.17.vectors)
uni.17.vectors2 <- left_join(data.frame(bac_sperm_17@sam_data), uni.17.vectors,  by = "Tube")
  
  un.pcoa.unifrac.17hrs.new <- ggplot(uni.17.vectors2, aes(x = Axis.1, y = Axis.2, fill = Trt, shape = Type)) +
  geom_point(size=4, alpha = 0.7) +
  scale_shape_manual(values = c(21,25)) +
  scale_fill_manual(values=cbbPalette) +
  xlab(paste("PcoA1 -", variation.axis1.17.uni, "%")) + 
  ylab(paste("PcoA2 -", variation.axis2.17.uni, "%")) +
  guides(fill=guide_legend(override.aes=list(shape=21))) +
  ggtitle(paste("17 Hours Postplanting")) +
  theme_bw() +
  theme(legend.position="bottom",
        legend.title=element_blank(),
        legend.key = element_blank(),
        plot.title=element_text(size=10, face = "bold"))
un.pcoa.unifrac.17hrs.new

#Dispersion for each Time-Point
beta.disp.17.uni <- betadisper(prok.dist.uni.17hrs, groups = c(bac_sperm_17@sam_data$Trt, bac_sperm_17@sam_data$Type))
permutest(beta.disp.17.uni)
```
### V2; Unweighted Unifrac #####
```{r}
bac_sperm_V2 <- BSpermosphere_RDS_CSS %>% 
  subset_samples(Time == "V2") %>% 
  phyloseq::filter_taxa(function(x) sum(x) > 0, TRUE)

# PERMANOVA - testing for differences in centroids
set.seed(12345)
prok.dist.uni.V2hrs = UniFrac(bac_sperm_V2, weighted = F) # create distance matrix
adonis.V2.uni <- adonis2(prok.dist.uni.V2hrs~Trt*Type, as(sample_data(bac_sperm_V2), "data.frame")) #Are there significant changes?
adonis.V2.uni
# The R2 is the variation due to different factors 
# pvalue is the pvalue for the factor

pvalue <- adonis.V2.uni$`Pr(>F)`[[1]]
R2 <- adonis.V2.uni$R2[[1]]

# ploting
ordination.unifrac.pcoa.V2hrs <- ordinate(bac_sperm_V2, "PCoA", distance = prok.dist.uni.V2hrs) # calculate the resemblance and ordinate using PCoA
uni.V2.values <- data.frame(ordination.unifrac.pcoa.V2hrs$values)

# percent variation
variation.axis1.V2.uni <- round(100*uni.V2.values$Relative_eig[[1]], 2)
variation.axis2.V2.uni <- round(100*uni.V2.values$Relative_eig[[2]], 2)

# data from ordinations
uni.V2.vectors <- data.frame(ordination.unifrac.pcoa.V2hrs$vectors) # positions of your points on the pcoa graph
uni.V2.vectors$Tube <- rownames(uni.V2.vectors)
uni.V2.vectors2 <- left_join(data.frame(bac_sperm_V2@sam_data), uni.V2.vectors,  by = "Tube")

  un.pcoa.unifrac.V2hrs.new <- ggplot(uni.V2.vectors2, aes(x = Axis.1, y = Axis.2, fill = Trt, shape = Type)) +
  geom_point(size=4, alpha = 0.7) +
  scale_shape_manual(values = c(21,25)) +
  scale_fill_manual(values=cbbPalette) +
  xlab(paste("PcoA1 -", variation.axis1.V2.uni, "%")) + 
  ylab(paste("PcoA2 -", variation.axis2.V2.uni, "%")) +
  guides(fill=guide_legend(override.aes=list(shape=21))) +
  ggtitle(paste("V2 Growth Stage")) +
  theme_bw() +
  theme(legend.position="bottom",
        legend.title=element_blank(),
        legend.key = element_blank(),
        plot.title=element_text(size=10, face = "bold"))
un.pcoa.unifrac.V2hrs.new

#Dispersion for each Time-Point
beta.disp.V2.uni <- betadisper(prok.dist.uni.V2hrs, bac_sperm_V2@sam_data$Trt)
permutest(beta.disp.V2.uni)
```
##### Figure 1; Stick all together #####
```{r}
Supplemental.1 <- ggpubr::ggarrange(un.pcoa.unifrac.0hrs.new,
                                       un.pcoa.unifrac.17hrs.new,
                                       un.pcoa.unifrac.V2hrs.new,
                                       sperm.0.ord.plot, 
                                       sperm.17.ord.plot,
                                       sperm.V2.ord.plot,
                                       sperm.0.ord.plot.bray,
                                       sperm.17.ord.plot.bray,
                                       sperm.V2.ord.plot.bray,
                                       labels = "auto",
                                       nrow = 2, ncol = 3, common.legend = T)
Figure1 <- ggpubr::ggarrange(pcoa.unifrac.0hrs.new, 
                                       pcoa.unifrac.17hrs.new,
                                       pcoa.unifrac.V2hrs.new,
                                       un.pcoa.unifrac.0hrs.new,
                                       un.pcoa.unifrac.17hrs.new,
                                       un.pcoa.unifrac.V2hrs.new,
                                    sperm.0.ord.plot.bray,
                                    sperm.17.ord.plot.bray,
                                    sperm.V2.ord.plot.bray,
                                       labels = "auto",
                                       nrow = 4, ncol = 3, common.legend = T)
```


```{r}
set.seed(12345)
adonis2(bac.bray~Trt*Time, as(sample_data(BSpermosphere_RDS_CSS), "data.frame"))

global.ord.bray <- phyloseq::ordinate(BSpermosphere_RDS_CSS, "MDS", "bray")
variation.axis1.bray <- round(100*global.ord.bray$values$Relative_eig[[1]], 2) # % variation on axis one
variation.axis2.bray <- round(100*global.ord.bray$values$Relative_eig[[2]], 2) # % variation on axis two

bac.ggplot.data.bray <- data.frame(global.ord.bray$vectors) # get the point values
bac.ggplot.data.bray$Tube <- rownames(bac.ggplot.data.bray) # add rownames
bac.ggplot.data.bray2 <- left_join(data.frame(cssbac.meta.data), bac.ggplot.data.bray,  by = "Tube") # THIS HAS TO BE THE SAME ON EACH FILE ##Purpose: add rest of metadata
bac.ggplot.data.bray2$Time.Point <- factor(bac.ggplot.data.bray2$Time, levels = c("Planting", "17hrs", "V2")) # order the time points in chronological order

bac.global.bray <- ggplot(bac.ggplot.data.bray2, aes(x = Axis.1, y = Axis.2, fill = Type, shape = Trt)) +
  facet_wrap(~factor(Time, levels = c("Planting", "17hrs", "V2")), scale = "free") +
  geom_point(size=4, alpha = 0.7)+
  scale_shape_manual(values=c(21,22, 23, 24,25,26,27), name = "Environment")+
  scale_fill_manual(values=cbbPalette, name = "Growth Stage")+
  xlab(paste("PcoA1 -", variation.axis1.bray, "%")) + 
  ylab(paste("PcoA2 -", variation.axis2.bray, "%")) +
  guides(fill=guide_legend(override.aes=list(shape= 21)),
         shape=guide_legend(override.aes=list(fill= "black")))+
  theme_classic() +
  ggtitle("Bray-Curtis")
bac.global.bray

bac.global.bray2 <- ggplot(bac.ggplot.data.bray2, aes(x = Axis.1, y = Axis.2, fill = Trt, shape = Time)) +
  #facet_wrap(~factor(Time, levels = c("Planting", "17hrs", "V2")), scale = "free") +
  geom_point(size=4, alpha = 0.7)+
  scale_shape_manual(values=c(21,22, 23, 24,25,26,27), name = "Growth Stage")+
  scale_fill_manual(values=cbbPalette, name = "Environment")+
  xlab(paste("PcoA1 -", variation.axis1.bray, "%")) + 
  ylab(paste("PcoA2 -", variation.axis2.bray, "%")) +
  guides(fill=guide_legend(override.aes=list(shape= 21)),
         shape=guide_legend(override.aes=list(fill= "black")))+
  theme_classic() +
  ggtitle("Bray-Curtis")
bac.global.bray2

```
##Fungal Bray Curtis
### Separate analysis based on time-points #
```{r}

#### 12 v. 18 hours
fun_sperm_0vs17 <- FSpermosphere_RDS_CSS %>% 
  subset_samples(Time %in% c("Planting", "17hrs", "V2")) %>% 
  phyloseq::filter_taxa(function(x) sum(x) > 0, TRUE)

# bray curtis distance
fun.bray.0vs17 = phyloseq::distance(fun_sperm_0vs17, "bray") # create bray-curtis distance matrix

# permanova
set.seed(12325)
adonis.output.0vs17 <- adonis2(fun.bray.0vs17~Trt*Time, as(sample_data(fun_sperm_0vs17), "data.frame"))
###adonis result there is a difference in Trt vs Time
adonis.output.0vs17.2 <- adonis2(fun.bray.0vs17~Type*Time, as(sample_data(fun_sperm_0vs17), "data.frame"))
###There is no relationship of Type x Time
adonis.output.0vs17.3 <- adonis2(fun.bray.0vs17~Type*Trt, as(sample_data(fun_sperm_0vs17), "data.frame"))
###There is a difference in Type x Trt (Sanity Check)

```
### 0 hours; Bray ####
```{r}
# filter to time point
fun_sperm_0.1 <- FSpermosphere_RDS_CSS %>% 
  subset_samples(Time == "Planting")%>%
  phyloseq::filter_taxa(function(x) sum(x) > 0, TRUE)
fun_sperm_0 <- fun_sperm_0.1 %>%
  subset_samples(Trt != "Control ") %>%
  phyloseq::filter_taxa(function(x) sum(x) > 0, TRUE)

# bray curtis distance
fun.bray.0 = phyloseq::distance(fun_sperm_0, "bray") # create bray-curtis distance matrix

# permanova
set.seed(12325)
adonis.output.0  <- adonis2(fun.bray.0~Trt*Type, as(sample_data(fun_sperm_0), "data.frame"))

pvalue0 <- adonis.output.0$`Pr(>F)`[[1]]
R2.0 <- adonis.output.0$R2[[1]]
```
###Dispersion for each Time-Point
```{r}
beta.disp.0 <- betadisper(fun.bray.0, fun_sperm_0@sam_data$Trt)

adonis.output.0
permutest(beta.disp.0)
sperm.0.ord <- phyloseq::ordinate(fun_sperm_0, "DCA", "bray")
Plot <- plot_ordination(fun_sperm_0, sperm.0.ord, color = "Trt", shape = "Type")
Plot


sperm.0.ord.vectors <- data.frame(sperm.0.ord$vectors)
sperm.0.ord.vectors$Tube <- rownames(sperm.0.ord.vectors)
sperm.0.ord.vectors2 <- left_join(data.frame(fun_sperm_0@sam_data), sperm.0.ord.vectors,  by = "Tube")
sperm.0.ord.values <- data.frame(sperm.0.ord$values)
```
### percent variation
```{r}
variation.axis1.0.bray <- round(100*sperm.0.ord.values$Relative_eig[[1]], 2)
variation.axis2.0.bray <- round(100*sperm.0.ord.values$Relative_eig[[2]], 2)

sperm.0.ord.plot <- ggplot(sperm.0.ord.vectors2, aes(x = Axis.1, y = Axis.2, fill = Trt, shape = Type)) +
  geom_point(size=4, alpha = 0.7) +
  scale_shape_manual(values = c(21,22)) +
  scale_fill_manual(values=cbbPalette) + 
  xlab(paste("PcoA1 -", variation.axis1.0.bray, "%")) + 
  ylab(paste("PcoA2 -", variation.axis1.0.bray, "%")) +
  ggtitle(paste("0 hrs")) +
  theme_bw() +
  theme(legend.position="bottom",
        legend.title=element_blank(),
        legend.key = element_blank(),
        plot.title=element_text(size=10, face = "bold")) 
sperm.0.ord.plot
```
### 17 hours; Weighted ####
#### filter to time point
```{r}
fun_sperm_17 <- FSpermosphere_RDS_CSS %>% 
  subset_samples(Time == "17hrs") %>% 
  phyloseq::filter_taxa(function(x) sum(x) > 0, TRUE)
fun_sperm_17 <- fun_sperm_17 %>%
  subset_samples(Trt != "Control ") %>%
  phyloseq::filter_taxa(function(x) sum(x) > 0, TRUE)

# bray curtis distance
fun.bray.17 = phyloseq::distance(fun_sperm_17, "bray") # create bray-curtis distance matrix

# permanova
set.seed(12325)
adonis.output.17  <- adonis2(fun.bray.17~Trt*Type, as(sample_data(fun_sperm_17), "data.frame"))

pvalue.17 <- adonis.output.17$`Pr(>F)`[[1]]
R2.17 <- adonis.output.17$R2[[1]]
```
###Dispersion for each Time-Point
```{r}
beta.disp.17 <- betadisper(fun.bray.17, fun_sperm_17@sam_data$Trt)

adonis.output.17
permutest(beta.disp.17)

sperm.17.ord <- phyloseq::ordinate(fun_sperm_17, "DCA", "bray")
Plot2 <- plot_ordination(fun_sperm_17, sperm.17.ord, color = "Trt", shape = "Type")
Plot2

sperm.17.ord.vectors <- data.frame(sperm.17.ord$vectors)
sperm.17.ord.vectors$Tube <- rownames(sperm.17.ord.vectors)
sperm.17.ord.vectors2 <- left_join(data.frame(fun_sperm_17@sam_data), sperm.17.ord.vectors,  by = "Tube")
sperm.17.ord.values <- data.frame(sperm.17.ord$values)
```
###percent variation
```{r}
variation.axis1.17.bray <- round(166*sperm.17.ord.values$Relative_eig[[1]], 2)
variation.axis2.17.bray <- round(166*sperm.17.ord.values$Relative_eig[[2]], 2)

sperm.17.ord.plot <- ggplot(sperm.17.ord.vectors2, aes(x = Axis.1, y = Axis.2, fill = Trt, shape = Type)) +
  geom_point(size=4, alpha = 0.7) +
  scale_shape_manual(values = c(21,22)) +
  scale_fill_manual(values=cbbPalette) +
  xlab(paste("PcoA1 -", variation.axis1.17.bray, "%")) + 
  ylab(paste("PcoA2 -", variation.axis2.17.bray, "%")) +
  guides(fill=guide_legend(override.aes=list(shape=21))) +
  ggtitle(paste("17 hrs")) +
  theme_bw() +
  theme(legend.position="bottom",
        legend.title=element_blank(),
        legend.key = element_blank(),
        plot.title=element_text(size=10, face = "bold"))
sperm.17.ord.plot
```
### V2 hours; Bray ####
#### filter to time point
```{r}
fun_sperm_V2 <- FSpermosphere_RDS_CSS %>% 
  subset_samples(Time == "V2" & Trt != "Control") %>% 
  phyloseq::filter_taxa(function(x) sum(x) > 0, TRUE)

fun_sperm_V2 <- fun_sperm_V2 %>%
  subset_samples(Trt != "Control ") %>%
  phyloseq::filter_taxa(function(x) sum(x) > 0, TRUE)

# bray curtis distance
fun.bray.V2 = phyloseq::distance(fun_sperm_V2, "bray") # create bray-curtis distance matrix

# permanova
set.seed(12325)
adonis.output.V2  <- adonis2(fun.bray.V2~Trt*Type, as(sample_data(fun_sperm_V2), "data.frame"))

pvalue.V2 <- adonis.output.V2$`Pr(>F)`[[1]]
R2.V2 <- adonis.output.V2$R2[[1]]
```
###Dispersion for each Time-Point
```{r}
beta.disp.V2 <- betadisper(fun.bray.V2, fun_sperm_V2@sam_data$Trt)

adonis.output.V2
permutest(beta.disp.V2)

sperm.V2.ord <- phyloseq::ordinate(fun_sperm_V2, "DCA", "bray")
Plot3 <- plot_ordination(fun_sperm_V2, sperm.V2.ord, color = "Trt", shape = "Type")#added
Plot3 #added

sperm.V2.ord.vectors <- data.frame(sperm.V2.ord$vectors)
sperm.V2.ord.vectors$Tube <- rownames(sperm.V2.ord.vectors)
sperm.V2.ord.vectors2 <- left_join(data.frame(fun_sperm_V2@sam_data), sperm.V2.ord.vectors,  by = "Tube")
sperm.V2.ord.values <- data.frame(sperm.V2.ord$values)
```
###percent variation
```{r}
variation.axis1.V2.bray <- round(100*sperm.V2.ord.values$Relative_eig[[1]], 2)
variation.axis2.V2.bray <- round(100*sperm.V2.ord.values$Relative_eig[[2]], 2)

sperm.V2.ord.plot <- ggplot(sperm.V2.ord.vectors2, aes(x = Axis.1, y = Axis.2, fill = Trt, shape = Type)) +
  geom_point(size=4, alpha = 0.7) +
  scale_shape_manual(values = c(21,24)) +
  scale_fill_manual(values=cbbPalette) +
  xlab(paste("PcoA1 -", variation.axis1.V2.bray, "%")) + 
  ylab(paste("PcoA2 -", variation.axis2.V2.bray, "%")) +
  theme(legend.position="bottom",legend.title=element_blank(),legend.key = element_blank()) +
  guides(fill=guide_legend(override.aes=list(shape=21))) +
  ggtitle(paste("V2 Growth Stage")) +
  theme_bw() +
  theme(legend.position="bottom",
        legend.title=element_blank(),
        legend.key = element_blank(),
        plot.title=element_text(size=10, face = "bold"))
sperm.V2.ord.plot
pcoa_fun.bray <- ggpubr::ggarrange(Plot,
                               Plot2,
                               Plot3,
                               labels = "auto",
                               nrow = 1, ncol = 3, common.legend = TRUE, legend = "bottom")
pcoa_fun.bray

#Manually change the legend on power-point
#yellow circles are steamed bulk soil
#Show PCOA plots in equal proportions in respects with the axis
#Manually put in P-Value cause coding it is not worth it
###
#Bray takes into account Presence absence and its abundance
#Jacaard is only presence absence
#unifrac Weighted = weighted by abundancies, unweighted = raw
```


#Steamed vs Nonsteamed Analysis#
#Relative Abundance#
##### Community Composition of top 20 most abundant OTUs ######
##### Prokaryote #####
```{r}
topx <- top_taxa(bac_sperm, n = 20)

bac.composition <- bac_sperm %>%
  subset_taxa(OTU %in% topx) %>%
  microbiome::transform("compositional") %>%
  psmelt() %>% 
  group_by(Trt, Time, Label) %>%
  mutate(Time = factor(Time, levels = c("Planting", "17hrs", "V2"))) %>%
  summarise(MeanRelAbund = mean(Abundance)) %>%
  arrange(-MeanRelAbund) %>%
  left_join(as.data.frame(tax_table(bac_sperm), by = "Label")) %>% 
  ggplot(aes(Type, MeanRelAbund, fill = Label)) +
  geom_bar(stat = "identity") +
  theme_classic() +
  #scale_x_discrete(labels=c('Soil no seed', 'Cotton Sp.', 'Soybean sp.'))+
  scale_fill_manual(values= c(cbbPalette, ibm.cbb, tol.cbb)) +
  scale_y_continuous(labels = scales::percent) +
  labs(x = "", y = "Relative abundance (%)",
       title = "Prokaryote") + 
  theme(axis.text.x = element_text(angle=45, hjust=1),
        legend.text = element_text(face = "italic", size = 5),
        legend.title = element_blank(),
        legend.key.size = unit(0.3, 'cm')) +
  facet_wrap(~Time, nrow = 1)
bac.composition
```

##### Fungi #####
```{r}
topx.fungi <- top_taxa(fungi_sperm, n = 20)

fung.composition <- fungi_sperm %>%
  subset_taxa(OTU %in% topx.fungi) %>%
  microbiome::transform("compositional") %>%
  psmelt() %>% 
  group_by(Crop, Time.Point, Label) %>%
  mutate(Time.Point = factor(Time.Point, levels = c("0", "6", "12", "18"))) %>%
  summarise(MeanRelAbund = mean(Abundance)) %>%
  arrange(-MeanRelAbund) %>%
  left_join(as.data.frame(tax_table(fungi_sperm), by = "Label")) %>% 
  ggplot(aes(Crop, MeanRelAbund, fill = Label)) +
  geom_bar(stat = "identity") +
  theme_classic() +
  scale_x_discrete(labels=c('Soil no seed', 'Cotton Sp.', 'Soybean sp.'))+
  scale_fill_manual(values= c(cbbPalette, ibm.cbb, tol.cbb)) +
  scale_y_continuous(labels = scales::percent) +
  labs(x = "", y = "Relative abundance (%)",
       title = "Fungi") + 
  theme(axis.text.x = element_text(angle=45, hjust=1),
        legend.text = element_text(face = "italic", size = 5),
        legend.title = element_blank(),
        legend.key.size = unit(0.3, 'cm')) +
  facet_wrap(~Time.Point, nrow = 1)
fung.composition
```
##### Compositional Figure ####
```{r}
compositional.fig <- ggpubr::ggarrange(bac.composition, 
                                       fung.composition,
                             labels = "auto",
                             nrow = 2, ncol = 1, align = "v")
```
#PCOA Plots#
```{r}
pcoa <- ggplot(points2, aes(x = Axis.1, y = Axis.2, color = Extraction_Type, shape = Time)) +
    theme_classic() +
    xlab(paste("PCoA1 -", axis1, "%")) +
    ylab(paste("PCoA2 -", axis2, "%")) +
    ggtitle(paste(var1)) +
    theme(plot.title = element_text(hjust = 0.5)) +
    geom_point(size = 4) +
    scale_color_manual(values = c("#D55E00", "#0072B2", "#332288", "#882255"))
  pcoa
```





























## Final figure 
fig1ab <- ggarrange(fig1a, fig1b, labels = "auto", nrow = 2)

ggarrange(fig1ab, fig1c, labels = c("", "c"), ncol = 2)











###### Supplemental Figure 1 ######
supp.fig.1 <- ggpubr::ggarrange(sequenced.mock.fungi, 
                                fungi.rare, 
                                decontaminate, 
                                read.dist, nrow = 2, ncol = 2, labels = c("a", "b", "c", "d"))

fungi.css.norm.filt <- fungi.css.norm %>% 
  subset_samples(Type %in% c("Spermosphere", "Soybean Epiphytes", "Rhizosphere")) %>%
  phyloseq::filter_taxa(function(x) sum(x) > 0, TRUE) # remove taxa not present in samples cut out

fungi.css.norm.filt@sam_data$TimeXTrt <- interaction(fungi.css.norm.filt@sam_data$Time, fungi.css.norm.filt@sam_data$Trt)

bac.bray = phyloseq::distance(fungi.css.norm.filt, "bray") # create bray-curtis distance matrix

fungi.css.norm@sam_data
# PERMANOVA - testing for differences in centroids
set.seed(12325)
adonis2(bac.bray~Trt*Time, as(sample_data(fungi.css.norm.filt), "data.frame"))

global.ord.bray <- phyloseq::ordinate(fungi.css.norm.filt, "NMDS", "bray")
p1 = plot_ordination(fungi.css.norm.filt, global.ord.bray, type="samples", color="TimeXTrt", shape = "Type") + 
  stat_ellipse()
variation.axis1.bray <- round(100*global.ord.bray$values$Relative_eig[[1]], 2) # % variation on axis one
variation.axis2.bray <- round(100*global.ord.bray$values$Relative_eig[[2]], 2) # % variation on axis two

bac.ggplot.data.bray <- data.frame(global.ord.bray$vectors) # get the point values
bac.ggplot.data.bray$Tube <- rownames(bac.ggplot.data.bray) # add rownames
bac.ggplot.data.bray2 <- left_join(data.frame(fungi.css.norm.filt@sam_data), bac.ggplot.data.bray,  by = "Tube") # add rest of metadata

global.bray <- ggplot(bac.ggplot.data.bray2, aes(x = Axis.1, y = Axis.2, fill = Time, shape = Trt)) +
  geom_point(size=4, alpha = 0.7)+
  scale_shape_manual(values=c(21,22, 23, 24), name = "Treatment")+
  scale_fill_manual(values=cbbPalette, name = "Environment")+
  xlab(paste("PcoA1 -", variation.axis1.bray, "%")) + 
  ylab(paste("PcoA2 -", variation.axis2.bray, "%")) +
  guides(fill=guide_legend(override.aes=list(shape= 21)),
         shape=guide_legend(override.aes=list(fill= "black")))+
  theme_bw() +
  ggtitle("Bray-Curtis")







